### PR TITLE
Added option for selection of grid size

### PR DESCRIPTION
**Pull request by ckarthik14**
The user can change the grid size between 4x4, 5x5, and 6x6.

### CODE REVIEW

**Review by vineetjc — COMMENTED** [review]
Instead of just changing the grid size, make a sort of relation between grid size and number of turns to make it of reasonable difficulty.

File: main.py [hunk 1]
```diff
@@ -114,21 +124,42 @@ def search_key():
...
+                    if maxturns>lowerLimitTurnCount:
                         maxturns=maxturns-1
 
+                elif incGridSizeRect.collidepoint(event.pos):
+                    if gridLength<6:
+                        gridLength=gridLength+1
+                        
+                        upperLimitTurnCount, lowerLimitTurnCount = getLimits(gridLength)
+
+                        print(upperLimitTurnCount)
```
Comment (vineetjc) [inline main.py]: Make sure to remove debug print statements from hereon, these aren't needed.

File: main.py [hunk 2]
```diff
@@ -71,16 +78,19 @@ def search_key():
     BG=pygame.image.load('Images/jail2.png')
     therm=pygame.image.load('Images/thermoLightBlu.png')
 
-    rects=[0 for i in range(25)] #list for boxes in the game
     blacklist=[] #stores numbers of the boxes clicked (already)
-    key_idx=random.randint(1,25) #randomize box with key
     turncount=0 #to check turn count
     maxturns = 6 #default number of turns allowed
+    maxsize = 6
```
Comment (vineetjc) [inline main.py]: This variable wasn't used at all.

**Review by vineetjc — COMMENTED** [review]
Note the feedback, but I've merged the PR. Well done!

File: main.py (hunk 1 above)
Comment (ckarthik14) [reply main.py:142]: Sure @vineetjc , I forgot to account for this print statement, will remove them next time.

**Pull request merged**